Release guide Improvements (#1961)
demisto/demisto-sdk: docs/release_guide.md

### Validation before release:
1) Make sure the **CHANGELOG.md** file is in order and is updated with all the changes in the current release.
2) Create a new release branch on the sdk repo, formatted as `X.X.X`, e.g. `1.0.0` (push the branch to the remote). 
3) Make sure that both `sdk-nightly` and `sdk-master` builds passed.

>    * If **new SDK commits** were pushed after the nightly tests had started, manually trigger the sdk nightly build again as written in steps 4 + 5. This will test thr sdk release branch on content branch.

Reviewer: ```suggestion
   * If **new SDK commits** were pushed after the nightly tests had started, manually trigger the sdk nightly build again as written in steps 4 + 5. This will test the sdk release branch with content master.
```

Author: it's a new content branch - in step 4 we should open a new branch and update the SDK version in the requirements file to the release version. The branch's content is identical to master... but it is still a different branch so I'm not sure what would be less confusing to write here.

>    * If **no new SDK commits** were done after the nightly tests, skip step 4 + 5.

Reviewer: ```suggestion
   * If **no new SDK commits** were done after the nightly tests, skip steps 4 and 5.
```
